core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataManager.java

```diff
@@ -1329,4 +1310,20 @@ public boolean isValid(Integer id) {
         }
         return true;
     }
+
+    public static boolean hasResult(SearchResponse searchResponse) {
+        return searchResponse.getHits().getTotalHits().value > 0;
+    }
+    SearchResponse searcherForReferencingMetadata(ServiceContext context, AbstractMetadata metadata) throws Exception {
+        GeonetContext gc = (GeonetContext) context.getHandlerContext(Geonet.CONTEXT_NAME);
+        EsSearchManager searchMan = gc.getBean(EsSearchManager.class);
+        StringBuilder query = new StringBuilder(String.format("xlink:*%s*", metadata.getUuid()));
+        query.append(String.format(" AND (%s:y OR %s:n)", IS_TEMPLATE, IS_TEMPLATE).toString());
+        return searchMan.query(query.toString(), null, 0, 0);
```

Comment:
I really don't know what happen when size is 0, as there are no tests, I assume totalHit can be greater than 0.

Reply:
it can be `0` or `> 0`, works in both case
but yes there is zero test in the whole class and I don't feel comfortable adding a test mocking the `searchManager`, but maybe it's doable.